Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Socket/AzSocket_UnixLike.cpp

```diff
@@ -294,8 +294,24 @@ namespace AZ::AzSock
         return HandleSocketError(::select(sock + 1, readfdsock, writefdsock, exceptfdsock, timeout));
     }
 
+#if defined(CARBONATED) && defined(AZ_PLATFORM_ANDROID)
+    AZ::s32 Poll(AZPOLLFD* pFDs, AZ::s32 countFDs, AZ::s32 timeoutMs)
+    {
+        return ::poll(pFDs, countFDs, timeoutMs);
+    }
+#endif
+
     AZ::s32 IsRecvPending(AZSOCKET sock, AZTIMEVAL* timeout)
     {
+#if defined(CARBONATED) && defined(AZ_PLATFORM_ANDROID)
+        // Use poll() instead select() for Android to allow the sockets with descriptor values >= FD_SETSIZE
+        AZPOLLFD fdPollData;
+        fdPollData.fd = sock;
+        fdPollData.events = POLLIN;
+        fdPollData.revents = 0;
+
+        return Poll(&fdPollData, 1, timeout ? timeout->tv_sec * 1000 + timeout->tv_usec / 1000 : -1) > 0 ? 1 : 0;
+#else
```

Comment:
Forgot to mention - O3DE really tries to minimize platform-specific guards like AZ_PLATFORM_ANDROID.  Given this is socket code, is it a change we can enable for all unix-like platforms?  Will this work for iOS & Linux?

If so, we should ditch the AZ_PLATFORM_ANDROID guards.  Also please make a PR back to O3DE.